### ISSUE TITLE
Converting a lot of *.d.ts files

### ISSUE THREAD

**Issue by ts2kt** [Kotlin/dukat#2]
There are such large projects, written directly in TypeScript and contains complex directory structure with a lot of files (hundreds). Such projects are compiled with option `--declaration` (which autogenerate `*.d.ts` for each `*.ts` files) and [published](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) directly to `npm`
Trying to convert typings of such project, I ran into a number of problems. Here are some of them:

1. Need option to specify directory, from which `dukat` will take `*.d.ts` files recursively and put result to destination directory keeping source directory structure. It could be resolved with some scripting, but...
2. I've tried to specify 200+ `*.d.ts` files via `$ dukat /path/to/**/*.d.ts`. It takes `> 1 hour` to convert them (2-4 files per minute)
3. Option `-b some_package_name` gave no effect. Each file get same package name as it file name. It also don't work for single file. I expect that I can specify same package name for all files. Ability to specify `@file:JsModule("path-to-module")` would be also useful.
4. Interface names like `'T$0'` will be duplicated, because each file converted independently.
5. If one `external class` implement interface from another file, there are no `override` modifier, because each file converted independently.
6. _Notice_: as mentioned [here](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) `package.json` could contain `types` property or `index.d.ts` at the root of the package. It would be nice if also it were ability to analyze dependencies starting from `index.d.ts` before just converting all files from npm package.
Also mainly `index.d.ts` contains re-exports.

**Comment by Schahen** [comment]
1. Will consider this as an enhancement however I still feel that this should be resolved on a caller side
2. I'll investigate. The bottleneck is the initialization  of javascript part (ts compiler plus our logic) - however this should happen only once per call. The translation  itself not supposed to take that long.
3. created a separate issue - https://github.com/Kotlin/dukat/issues/8
4. Will be considered as a feature request (reuse entities between files) 
5.  A separate issue - https://github.com/Kotlin/dukat/issues/9
6. Will be considered as a feature request.

**Comment by webleaf** [comment]
4. Are there any plans for this?
5. I see the milestone `0.0.5` was removed for #9 

`dukat` is not useful without ability to convert multiply files (it requires a huge amount of manual work after `dukat`, because many files are related).

**Comment by Schahen** [comment]
@webleaf There are definitely plans for this, I've removed #9 from 0.0.5 only because 0.0.5 is a bugfix release which is planned to happen tomorrow (so earlier than next Friday). I'm also reviewing all that can be improved regarding the import/compile-all-togeher etc. issues and than I'll create tasks accordingly and we'll see what's next. Since the process is pretty transparent you'll see related issues pinned to corresponding milestone )

**Comment by ts2kt** [comment]
> 4. Will be considered as a feature request (reuse entities between files)
4. As a temporary solution, it could be global counter for all specified files. This will avoid the name conflicts in the package (otherwise a package contains many `'T$0'`, `'T$1'` etc).
